### PR TITLE
Fix #3866 - includeTables, improve filtering and lazy connection

### LINKED ISSUE #3866: Service Connection & Schema Name - SourceConfig includeTables
We are missing the includeTables property in the JSON Schema for the source config of the metadata ingestion. Add it and put it back to the sql_source.py

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #3866

- Fixed table and schema filtering not using the centralised methods
- SQL source connection is now lazy

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by ayush-shah — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 1]
```diff
@@ -184,6 +186,16 @@ def session(self) -> Session:
 
         return self._session
 
+    @property
+    def connection(self) -> Connection:
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/sql_source.py:190]: we only use this connection for generate_sample_data we should move this into that method and only get it if the generate_sample_data is set to True

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:190]: I use this connection for the profiler as well to not create more than one. Felt a good call to have it specific to the source. Shall we move that?

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:190]: ok actually I use the engine not the connection, got confused. I'll move this. Thanks

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:190]: 🤔

**Review by harshach — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/sql_source.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:190]: the connection will get initialised only on call. If sampleData is false, nothing will run here. Thanks 🙏

**Pull request merged**